Tribler/community/triblerchain/community.py

```diff
@@ -183,76 +183,116 @@ def get_node(self, public_key, nodes, total_up=None, total_down=None, total_neig
         edges = {}
 
-        for edge in self.persistence.get_graph_edges(public_key, neighbor_level):
+        query_result = self.persistence.get_graph_edges(public_key, neighbor_level)
+
+        # Find all nodes and all edges in the result
+        for edge in query_result:
             from_pk = str(edge[0])
             to_pk = str(edge[1])
             amount_up = edge[2]
             amount_down = edge[3]
             nodes[from_pk] = self.get_node(from_pk, nodes, total_up=edge[4], total_down=edge[5],
                                            total_neighbors=edge[6])
             nodes[to_pk] = self.get_node(to_pk, nodes)
-            self.update_edges(from_pk, to_pk, edges, amount=amount_up)
-            self.update_edges(to_pk, from_pk, edges, amount=amount_down)
+            if from_pk not in edges:
+                edges[from_pk] = []
+            edges[from_pk].append((to_pk, amount_up, amount_down))
+            if to_pk not in edges:
+                edges[to_pk] = []
+            edges[to_pk].append((from_pk, amount_down, amount_up))
+
+        return nodes, edges
+
+    @blocking_call_on_reactor_thread
+    def get_graph(self, public_key, neighbor_level, max_neighbors, mandatory_nodes):
+        """
+        Return a dictionary with the neighboring nodes and edges of a certain focus node within a certain radius,
+        regarding the local TrustChain database, limited in the amount of higher level neighbors per node.
+
+        :param public_key: the public key of the focus node in raw format
+        :param neighbor_level: the radius within which the neighbors have to be returned
+        :param max_neighbors: the maximum amount of higher level neighbors one node may have
+        :param mandatory_nodes: list of nodes that have to be in the visualization if possible
+        :return: a tuple of a list with nodes and a list with edges
+        """
+        nodes, edges = self.format_edges(public_key, neighbor_level)
 
-        return_nodes = nodes.values()
+        return_nodes = []
         return_edges = []
 
+        nodes_visited = set()
+        this_level = set()
+        this_level.add(public_key)
+        # If the focus is in the list, always add it to the view
+        # if own_node in nodes:
+        #     this_level.add(own_node)
+        next_level = set()
+
+        # Check if the list of edges is empty
+        if not edges:
+            return_nodes.append({"public_key": public_key, "total_up": nodes[public_key]["total_up"],
+                                 "total_down": nodes[public_key]["total_down"],
+                                 "total_neighbors": nodes[public_key]["total_neighbors"]})
+            return_nodes[0]["score"] = calculate_score(return_nodes[0])
+            return return_nodes, return_edges
+
+        # Limit the number of higher-level neighbors per node in the graph
+        for current_level in range(neighbor_level + 1):
+            for node in this_level:
```

Comment:
`this_level` is updated each iteration. It only contains 1 value the first iteration (unless the chosen node only has 1 member of course).